src/compiler.rs

```diff
@@ -340,7 +341,16 @@ Path 2: {:?}",
 
                 if self.print_asm {
                     // prints the assembly
+
                     for l in &final_asm {
+                        // prints the function's path as well if it's a function call
+                        if l.contains("____") {
+                            let mut fn_name = l.split("____").next().unwrap();
+                            fn_name = fn_name.strip_prefix("call ").unwrap_or(fn_name);
+                            println!("{} ({})", l, self.state.fn_to_path.get(fn_name).unwrap());
```

Comment:
This line crashes in certain cases. The functions in the compiled tasm source are not guaranteed to match 1:1 with ashlang sources. For example, blocks and loops were both implemented as functions in tasm so that they could easily be conditionally jumped to. 

This line crashes in the [if_assignment_test function](https://github.com/chancehudson/ashlang/blob/main/test-vectors/if_assignment_test.ash). You can run the tasm test cases using `sh ./test.sh`.

I'd say this type of logic should be implemented in `tasm/vm.rs`. Triton assembly supports comments so this logic could be included in the compiled output as well. tasm comments start with `//`

Reply:
This logic has been removed. Simply calling `fn_to_path()` was sufficient during the ASM construction steps:
https://github.com/chancehudson/ashlang/pull/49/files#diff-19a5e6a0406f9184fd74dba07991bee321e608450fff0e30117f02c49c97ea8aR313-R317

Reply:
Nice, this is a good solution